### PR TITLE
On hold applications tab

### PR DESCRIPTION
**Pull request by wevekamola**
In PR #290 `On hold` tab has been added to the application dashboard and also `Active` tab has been renamed to `Open` and `Rejected` tab has been renamed to `Closed`

### CODE REVIEW

File: app/Models/HR/Application.php [hunk 1]
```diff
@@ -150,6 +153,22 @@ public function scopeNonRejected($query)
         return $query->where('status', '!=', config('constants.hr.status.rejected.label'));
     }
 
+    /**
+     * get applications where status is new and in-progress
+     */
+    public function scopeOpenApplication($query)
```
Comment (rathorevaibhav) [inline app/Models/HR/Application.php]:
```
Application::with('applicationRounds')->openApplication()->get();
```
has repetitive term `application`. 

If we rename the scope to scopeIsOpen then the following query is more readable and meaningful.
```
Application::with('applicationRounds')->isOpen()->get();
```

File: app/Models/HR/Application.php [hunk 2]
```diff
@@ -93,8 +93,11 @@ public function scopeFilterByStatus($query, $status)
             case config('constants.hr.status.rejected.label'):
                 $query->rejected();
                 break;
+            case config('constants.hr.status.on-hold.label'):
+                $query->onHold();
+                break;    
             default:
-                $query->nonRejected();
```
Comment (rathorevaibhav) [inline app/Models/HR/Application.php]: Now that we don't need nonRejected scope anymore, I think we can remove it altogether?

File: app/Models/HR/Application.php [hunk 3]
```diff
@@ -150,6 +153,22 @@ public function scopeNonRejected($query)
...
+     */
+    public function scopeOpenApplication($query)
+    {
+        return $query->whereIn('status', array(config('constants.hr.status.new.label'), config('constants.hr.status.in-progress.label')));
+    }
+
+    /**
+     * get applications where status is on-hold
+     */
+    public function scopeOnHold($query)
```
Comment (rathorevaibhav) [inline app/Models/HR/Application.php]: 👍

File: app/Models/HR/Application.php (hunk 2 above)
Comment (wevekamola) [reply app/Models/HR/Application.php]: @rathorevaibhav We are using `nonRejected` scope in other files also `Applicant.php` and `ApplicationController.php` i don't think it's a good idea to remove it. Let me know your thoughts.

File: app/Models/HR/Application.php (hunk 1 above)
Comment (wevekamola) [reply app/Models/HR/Application.php]: 👍

File: app/Models/HR/Application.php (hunk 2 above)
Comment (wevekamola) [reply app/Models/HR/Application.php]: @rathorevaibhav Yeah we should only count `new ` and `in-progress` as open application. Is that what you are saying?

Comment (rathorevaibhav) [reply app/Models/HR/Application.php]: Yes. openApplications = `new` and `in-progress` applications

That means, there's no section where we need to show nonRejected (earlier openApplications = `nonRejected`). So we can remove nonRejected from everywhere.

**Pull request merged**